contracts/PixelmonPartySquad/PxPartySquad.sol

```diff
@@ -0,0 +1,392 @@
+            week.distributions[week.treasureCount].totalSupply = _treasureCounts[index];
+            week.remainingSupply += _treasureCounts[index];
+        }
+    }
+
+    /// @notice Set a list of winners for a particular week
+    /// @param _weekNumber The current week number
+    /// @param _winners List of wallet addresses that have been selected as winners
+    /// @param _treasureCounts Amount of treasure that have been awarded to the corresponding winner
+    function updateWeeklyWinners(
+        uint256 _weekNumber,
+        address[] memory _winners,
+        uint8[] memory _treasureCounts
+    ) external onlyModerator(msg.sender) validArrayLength(_winners.length, _treasureCounts.length) validWinnerUpdationPeriod(_weekNumber) {
+        for (uint256 index = 0; index < weekInfos[_weekNumber].specialTreasureWinners.length; index++) {
+            address specialTreasureWinner = weekInfos[_weekNumber].specialTreasureWinners[index];
+            weekInfos[_weekNumber].specialTreasureWinnerMap[specialTreasureWinner] = false;
+        }
+        uint256 randomNumber = getRandomNumber();
+        uint256 randomIndex = randomNumber - ((randomNumber / _treasureCounts.length) * _treasureCounts.length);
+        uint256 counter = 0;
+        uint256 specialTreasureWinnerCount = 0;
+        uint256 treasureCount = 0;
+        address[] memory tmpSpecialTreasureWinners = new address[](weekInfos[_weekNumber].specialTreasureCount);
+        while (counter < _treasureCounts.length) {
+            if (randomIndex == _treasureCounts.length) {
+                randomIndex = 0;
+            }
+            if (
+                specialTreasureWinnersLimit[_winners[randomIndex]] < maxSpecialTreasureLimit &&
+                specialTreasureWinnerCount < weekInfos[_weekNumber].specialTreasureCount &&
+                _treasureCounts[randomIndex] > 0
+            ) {
+                weekInfos[_weekNumber].specialTreasureWinnerMap[_winners[randomIndex]] = true;
+                tmpSpecialTreasureWinners[specialTreasureWinnerCount] = _winners[randomIndex];
+                specialTreasureWinnerCount++;
+            }
+
+            weekInfos[_weekNumber].winners[_winners[randomIndex]].claimLimit = _treasureCounts[randomIndex];
+            treasureCount += _treasureCounts[randomIndex];
+            unchecked {
+                randomIndex++;
+                counter++;
+            }
+        }
+        if (treasureCount > weekInfos[_weekNumber].remainingSupply + weekInfos[_weekNumber].specialTreasureCount) {
+            revert("Invalid Treasure Amount");
+        }
+
+        weekInfos[_weekNumber].specialTreasureWinners = tmpSpecialTreasureWinners;
+        emit WeeklyWinnersSet(_weekNumber, tmpSpecialTreasureWinners);
+    }
+
+    /// @notice Add a list of wallet addresses that have won Special Treasure
+    /// @param _previousWinners List of addresses that have won Special Treasure
+    /// @param _counts 'true' means already won
```

Comment:
I think the comment needs to be rephrased.
`counts` has `uint256[]` type but the comment says _"@param _counts 'true' means already won"_
Could you clarify it means and how we suppose to set the value?